xarray/conventions.py

```diff
@@ -837,11 +838,12 @@ def decode_cf_variables(variables, attributes, concat_characters=True,
 
     def stackable(dim):
         # figure out if a dimension can be concatenated over
-        if dim in variables:
+        if dim in variables and dim not in concat_dimensions:
```

Comment:
It probably makes more sense to simply use `concat_dimensions` instead of the heuristics here if it is provided? E.g., `if concat_dimensions is not None: return dim in stack_dimensions`?